dbms/src/DataStreams/ParallelParsingBlockInputStream.h

```diff
@@ -193,8 +193,10 @@ class ParallelParsingBlockInputStream : public IBlockInputStream
         pool.scheduleOrThrowOnError(std::bind(&ParallelParsingBlockInputStream::parserThreadFunction, this, unit_number));
     }
 
-    void waitForAllThreads()
+    void finishAndWait()
     {
+        finished.store(true, std::memory_order_release);
```

Comment:
Why do you use *std::memory_order_release* there and don't use *std::memory_order_acquire* somewhere else? Also I see *finished = true* in other part of code where default sequential consistency model is used.